### PR TITLE
ocs operator e2e CI job

### PR DESCRIPTION
**Pull request by davidvossel**
This PR does the following

* Builds container images for ocs-operator and ocs-registry
* Adds an AWS e2e test job that deploys and validates the ocs-operator on an AWS cluster.
* Adds a Promote postsubmit job which will allow us to create a ocp periodic job later on to validate ocs-operator as new ocp releases land.

### CODE REVIEW

**Comment by davidvossel** [comment]
We'll need the deploy scripts and make targets required by the Openshift CI job to be merged into the ocs-operator src tree before we can test this.

https://github.com/openshift/ocs-operator/pull/91

**Comment by davidvossel** [comment]
> We'll need the deploy scripts and make targets required by the Openshift CI job to be merged into the ocs-operator src tree before we can test this.
> openshift/ocs-operator#91

or is there a way we can test these changes against a specific ocs-operator PR?

File: ci-operator/jobs/openshift/ocs-operator/openshift-ocs-operator-master-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            cpu: 10m
+        volumeMounts:
+        - mountPath: /etc/sentry-dsn
+          name: sentry-dsn
+          readOnly: true
+      serviceAccountName: ci-operator
+      volumes:
+      - name: sentry-dsn
+        secret:
+          secretName: sentry-dsn
```
Comment (kshlm) [inline ci-operator/jobs/openshift/ocs-operator/openshift-ocs-operator-master-postsubmits.yaml]: If I understand correctly, this postsubmit job creates operator and registry images. Will these be available for use outside the CI? If no, why are we doing this?

File: ci-operator/config/openshift/ocs-operator/openshift-ocs-operator-master.yaml [hunk 2]
```diff
@@ -1,16 +1,56 @@
...
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/openshift/ocs-operator/build/_output/bin/ocs-operator
+  dockerfile_path: openshift-ci/Dockerfile.deploy
+  to: ocs-operator
+- from: operator-registry
+  dockerfile_path: openshift-ci/Dockerfile.registry.intermediate
+  to: operator-registry-base
```
Comment (kshlm) [inline ci-operator/config/openshift/ocs-operator/openshift-ocs-operator-master.yaml]: I see the intermediary Dockerfile being used here. Why do we need to have a intermediary image for the registry image?

Comment (davidvossel) [reply ci-operator/config/openshift/ocs-operator/openshift-ocs-operator-master.yaml]: this link explains it well. https://github.com/redhat-developer/devconsole-operator/tree/master/openshift-ci#operator-registry-base

The intermediate image is part of a trick used to be able to accurately predict the ocs-operator container image url that is being built by this same job. The intermediate build has the unique `$OPENSHIFT_BUILD_NAMESPACE` propagated into it that is related to this CI job. By basing the ocs-registry container off of the intermediate container, we are able to use that `OPENSHIFT_BUILD_NAMESPACE` var to predict the url of the ocs-operator image which is injected into the CSV. 

Basically, it's just a way to have the ocs-registry CSV reference the ocs-operator image correctly. otherwise the ocs-registry would point to the quay.io/ocs-dev/ocs-operator:latest image instead of the ocs-operator container built by this CI job.

Also, I didn't come up with this pattern. Several other projects are using this method to solve the exact same issue.

Comment (kshlm) [reply ci-operator/config/openshift/ocs-operator/openshift-ocs-operator-master.yaml]: Thanks for the explanation! I'd assumed this to be a pattern used elsewhere, but wanted to know why.

**Comment by davidvossel** [comment]
> imagestreamtags.image.openshift.io "ubi:8" not found

it looks like i need to fix something in the config. i'll update this shortly.

**Comment by davidvossel** [comment]
> it looks like i need to fix something in the config. i'll update this shortly.

i updated to use ubi7 as the base container, we'll see if that gets us any further

**Comment by davidvossel** [comment]
```No resources found.
Waiting for deployment ocs-operator to be created...
No resources found.
Waiting for deployment ocs-operator to be created...
No resources found.
Waiting for deployment ocs-operator to be created...
No resources found.
Waiting for deployment ocs-operator to be created...
No resources found.
Waiting for deployment ocs-operator to be created...
make: *** [cluster-deploy] Error 1
Timed out waiting for deployment ocs-operator to be created```

looks like the test times out waiting for the deployment to complete.

I have enough information to attempt to debug this locally. Looking at the images, they are getting created properly which is great at least.  It's possible ocs-operator master might not actually be able to be deployed successfully at the moment for some reason. I'll figure it out :)

**Comment by davidvossel** [comment]
e2e test passes now

before Merging this, please merge [this PR](https://github.com/openshift/ocs-operator/pull/94) into master. I noticed that we needed to increase the deploy timeout. Occasionally the olm operator doesn't reconcile the subscription immediately causing a delay in setup.
